### PR TITLE
Some text input changes

### PR DESCRIPTION
**Pull request by rmsthebest**
Hi,
I'm not sure if you are even interested in changes, but when I was trying out your library I noticed that backspace wasn't quite working. 

I tried to copy your code patterns to not be intrusive, even if some of the code looks a bit repetitive with my additions.

### CODE REVIEW

**Review by jkb0o — COMMENTED** [review]
Wow! What a gift! Thanks a lot for your contribution. I'm definitely interested in changes, especially in such a useful ones.

I'll try your code later today, but I'd like to notice some things already.

The first thing is about CHAR_BACKSPACE. I'm using Mac OSX and my keyboard has a key looks exactly like a backspace and behaves exactly like backspace. But I still receive DELETE character when pressing it. I guess you can left this part as is and I'll make OS-related adjustments for it.

The second notice is about `keyboard.pressed(KeyCode::Right)`. As far as I understand the code, `pressed(..)` returns `true` every time system executes if hold the arrow key down. In this case cursor movement speed depends on the frame rate while you holding the key. This is not the behaviour I'd like to see here. The solution is to introduce some `Local<...>` variable that accumulates the time key is pressed and change the selection/cursor position in controllable way, so move speed doesn't depend on the frame rate. Correct me If I wrong, can't proof anything yet. But If I right, it would be cool to have such accumulator on this system.

And repeating is ok for me here. I believe the code would be more readable if `shift` & `!shift` was exposed into separate top-level if-else branches for a hole keyboard input, but for now It is just ok.

**Comment by rmsthebest** [comment]
the macos fix is kind of fragile since it relies on the if statement of backspace being before the delete. Hopefully that's okay since the current method will not be able to differentiate between the two anyway.
Maybe it could just use KeyCode::Back and KeyCode::Delete? I don't know what works on your machine I'll leave that to you :)

File: crates/belly_widgets/src/input/text.rs [hunk 1]
```diff
@@ -271,12 +310,26 @@ fn process_keyboard_input(
...
+                    }
+                }
             } else {
                 if !selected.is_empty() {
                     chars.drain(selected.range());
                     index = selected.min;
                     selected.stop();
                 }
+                // this will try to insert anything, even if it doesnt have a textual representaion
+                // (escape for exampe will be inserted as a space) some constraint would be useful
```
Comment (jkb0o) [inline crates/belly_widgets/src/input/text.rs:332]: This is a good point. I'd appreciate if you apply a fix for this part, something like replacing `} else {` with 
```rust
// LAST_CONTROL_CHARACTER is '\u{1f}'
} else if ch.char > LAST_CONTROL_CHARACTER {
```

**Review by jkb0o — CHANGES_REQUESTED** [review]
Looks good to me! Just tried the code, everything work as expected. OSX fix looks nice as well.

Can you please resolve the conflicts and may be drop the commit with bevy version update? I just pushed this one to the mainstream (I'd like to squash your PR into a single commit).

**Comment by rmsthebest** [comment]
well this has been an adventure. I've never dropped a commit before, I think it worked out though. Thanks for being patient with me

**Review by jkb0o — APPROVED** [review]
Great job! Thank you so much!

**Comment by jkb0o** [comment]
@rmsthebest My apologises, I was wrong about timer. It wasn't required. One of condition to not exit system early is non-empty keyboard events stream. bevy emits `pressed` events while you hold key pressed based on os preferences. I rethought how input should be handled and came to solution in 6ce07af. Without your PR, this would not have happened soon. Thank you!

**Pull request merged**